CodeStarterCamp_Week4/FitnessCenter.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  FitnessCenter.swift
+//  CodeStarterCamp_Week4
+//
+//  Created by 정선아 on 2022/05/10.
+//
+
+import Foundation
+
+enum InputError: Error {
+    case empty
+    case outOfValue
+    case invaildValue
+}
+
+class FitnessCenter {
+    let name: String
+    var targetBodyCondition: BodyCondition
+    var member: Person?
+    var listOfRoutine: [Routine]
+    
+    init(name: String, targetBodyCondition: BodyCondition, member: Person, listOfRoutine: [Routine]) {
+        self.name = name
+        self.targetBodyCondition = BodyCondition()
+        self.member = member
+        self.listOfRoutine = listOfRoutine
+        
+    }
+    
+    convenience init(name: String, listOfRoutine: [Routine]) {
+        self.init(name: name, targetBodyCondition: BodyCondition(), member: Person(name: "", bodyCondition: BodyCondition()), listOfRoutine: listOfRoutine)
+    }
+    
+    func exerciseRoutine() {
+            do {
+                self.member?.name = try writeName()
+                self.targetBodyCondition = try inputBodyCondition()
+            } catch InputError.empty {
+                print("입력되지 않았습니다. 다시 입력해주세요.")
+            } catch InputError.invaildValue {
+                print("없는 번호입니다. 다시 입력해주세요.")
+            } catch InputError.outOfValue {
+                print("값의 갯수가 적거나 큽니다.")
+            } catch {
+                print("알 수 없는 오류입니다. 다시 시도해주세요.")
+            }
+        tryRoutine()
+        }
```

Comment:
- member 프로퍼티는, 아직 존재하지 않는 옵셔널 상태잖아요?
그러니까, 옵셔널 값에 접근해서 name 을 건드리는 것 보다는
그냥 통째로 `Person 인스턴스를 생성해서 할당`하는 게 좋아보여요. 👍🏻 
(코드를 보면서 이해해보세요!)

- 그리구 전반적으로, `들여쓰기`가 좀 과하게 된 코드들이 종종 보입니다!
이 메서드도 들여쓰기가 스페이스바 8번이나 되어 있어요! (보통 4번임)

```suggestion
    func exerciseRoutine() {
        do {
            let memberName = try writeName() // 유저의 이름을 상수로 받아낸 뒤에
            member = Person(name: memberName, bodyCondition: BodyCondition()) // 이니셜라이저에서 사용
            bodyGoal = try inputBodyGoal() // 이건 아래서 다룰게요.
        } catch InputError.empty {
            print("입력되지 않았습니다. 다시 입력해주세요.")
        } catch InputError.invaildValue {
            print("없는 번호입니다. 다시 입력해주세요.")
        } catch InputError.outOfValue {
            print("값의 갯수가 적거나 큽니다.")
        } catch {
            print("알 수 없는 오류입니다. 다시 시도해주세요.")
        }
        tryRoutine()
    } // 줄 잘 정렬하기
```

Reply:
- 들여쓰기는 아마 위에서도 말했지만 플레이그라운드에서 코드를 복붙해서 가져오면서 생긴 오류(?) 같아요... 
한 번 찾아봐야겠네요...ㅠㅠ

- 이니셜라이저를 이런 식으로도 사용할 수 있군요! 무조건 init() 메서드 안에서만 사용해야하는 줄 알았어요 ㅇ0ㅇ!!